# Домашнє завдання до теми "Основи роботи з рядками та математичні операції"

1. Створіть змінні `a`, `b`, `c` та `d`, задайте їм значення відповідно 101, 210, 111, 402, та обчисліть з допомогою Python наступний вираз:  $$ \frac{(a+b) * c}{d} $$

**Важливо**: тут ми маємо обчислити значення саме з використанням змінних, а не просто чисел, як би ми робили у калькуряторі. Змінні потрібні аби за бажання ми могли змінити їх значення та не переписувати код обчислення виразу, бо він буде працювати для різних значень змінних.

In [8]:
a = 101
b = 210
c = 111
d = 402
(𝑎 + 𝑏)*c / d


85.8731343283582

Очікувана відповідь: 85.8731343283582

2. Якою буде остача від ділення числа 11283 на 89? Обчисліть з допомогою Python.

In [9]:
11283 % 89

69

3. Скільки цілих разів поміститься число 89 в 11283? (Або що те саме: який буде результат цілочисельного ділення 11283 на 89?)

In [10]:
11283 //89

126

4. Запишіть у змінну `name` ваше імʼя.

In [17]:
name = "Yulia"
name


'Yulia'

5. Запишіть у змінну `surname` ваше прізвище.

In [18]:
surname = "Matiyuk"
surname

'Matiyuk'

6. Сконкатенуйте імʼя та прізвище в одну фразу через пробіл. Запишіть результат у нову змінну `name_surname` та виведіть результат на екран.  

In [24]:
name + " " + surname

'Yulia Matiyuk'

Очікувана відповідь (тільки з вашими даними): 'Hanna Pylieva'

In [28]:
name_surname = "Yulia Matiyuk"
name_surname

'Yulia Matiyuk'

7. Зробіть усі літери змінної `name_surname` заглавними використовуючи метод рядків в Python:

In [3]:
s = 'Yulia Matiyuk'

In [4]:
print (s.upper())

YULIA MATIYUK


Очікувана відповідь (тільки з вашими даними): 'HANNA PYLIEVA'

8. Виведіть третю літеру вашого `name_surname`. Тут уважно, треба саме третя літера. Згадайте, з якого значення починається індексування значень в Python.

In [7]:
s[3]

'i'

Очікувана відповідь (тільки з вашими даними): 'n'

9. Виведіть першу та останню літеру вашого прізвища.

In [8]:
name_surname = 'Yulia Matiyuk'
len (name_surname)


13

In [9]:
name_surname [0]

'Y'

In [11]:
name_surname [12]

'k'

Очікувана відповідь (тільки з вашими даними): ('P', 'a')

10. Виведіть довжину вашого прізвища.

In [9]:
s = "Yulia Matiyuk"
len (s)

13

Очікувана відповідь (тільки з вашими даними): 7


11. Виведіть кількість літер `a` у столиці Мадагаскару `Antananarivo`. Важливо врахувати всі літери, як великі, так і маленькі.

In [25]:
s = "Antananarivo"
len (s)

12

In [42]:
s.count("a")

3

In [43]:
s.count("A")

1

Очікувана відповідь: 4

12. Створіть рядок `my_string` зі значенням: "The quick brown fox jumps over the lazy dog!!!".   
Перевірте, чи починається рядок `my_string` з "The" і чи закінчується він "dog!". Виведіть результати на екран.

In [16]:
my_string = 'The quick brown fox jumps over the lazy dog!!!.'


In [17]:
print (my_string.startswith("The"))
print (my_string.endswith("dog!"))

True
False


13. Розділіть рядок `my_string` на список окремих слів, а потім з'єднайте всі слова через нижнє підкреслювання. Виведіть результат на екран.

In [19]:
my_string = 'The quick brown fox jumps over the lazy dog!!!.'
my_string.split(" ")

['The', 'quick', 'brown', 'fox', 'jumps', 'over', 'the', 'lazy', 'dog!!!.']

In [23]:
print ("_".join(['The', 'quick', 'brown', 'fox', 'jumps', 'over', 'the', 'lazy', 'dog!!!.']))

The_quick_brown_fox_jumps_over_the_lazy_dog!!!.


Очікуваний результат: `The_quick_brown_fox_jumps_over_the_lazy_dog!!!`